packages/core/admin/admin/src/pages/SettingsPage/pages/Webhooks/EditView/components/Events/index.js

```diff
@@ -0,0 +1,288 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+
+import { useFormikContext } from 'formik';
+import { useIntl } from 'react-intl';
+import styled from 'styled-components';
+import { FieldLabel, Flex, Typography, BaseCheckbox, Checkbox } from '@strapi/design-system';
+
+import { useModels } from '../../../../../../../hooks';
+
+export const formatValue = (value) =>
+  value.reduce((acc, curr) => {
+    const key = curr.split('.')[0];
+
+    if (!acc[key]) {
+      acc[key] = [];
+    }
+    acc[key].push(curr);
+
+    return acc;
+  }, {});
+
+const StyledTable = styled.table`
+  td {
+    height: ${52 / 16}rem;
+    width: 10%;
+    vertical-align: middle;
+    text-align: center;
+  }
+
+  tbody tr:nth-child(odd) {
+    background: ${({ theme }) => theme.colors.neutral100};
+  }
+
+  tbody tr td:first-child {
+    padding-left: ${({ theme }) => theme.spaces[7]};
+  }
+`;
+
+const getCEHeaders = (isDraftAndPublish) => {
+  if (isDraftAndPublish) {
+    return [
+      { id: 'Settings.webhooks.events.create', defaultMessage: 'Create' },
+      { id: 'Settings.webhooks.events.update', defaultMessage: 'Update' },
+      { id: 'app.utils.delete', defaultMessage: 'Delete' },
+      { id: 'app.utils.publish', defaultMessage: 'Publish' },
+      { id: 'app.utils.unpublish', defaultMessage: 'Unpublish' },
+    ];
+  }
+
+  return [
+    { id: 'Settings.webhooks.events.create', defaultMessage: 'Create' },
+    { id: 'Settings.webhooks.events.update', defaultMessage: 'Update' },
+    { id: 'app.utils.delete', defaultMessage: 'Delete' },
+  ];
+};
+
+const getCEEvents = (isDraftAndPublish) => {
+  if (isDraftAndPublish) {
+    return {
+      entry: ['entry.create', 'entry.update', 'entry.delete', 'entry.publish', 'entry.unpublish'],
+      media: ['media.create', 'media.update', 'media.delete'],
+    };
+  }
+
+  return {
+    entry: ['entry.create', 'entry.update', 'entry.delete'],
+    media: ['media.create', 'media.update', 'media.delete'],
+  };
+};
```

Comment:
What do you think about reversing that? Return the second array always and extend it with draft and publish if the feature is enabled? This way you don't have to maintain two arrays.